GVFS/GVFS.Service/Handlers/GetActiveRepoListHandler.cs

```diff
@@ -42,14 +43,7 @@ public void Run()
                 {
                     if (!this.IsValidRepo(repoRoot))
                     {
-                        if (!this.registry.TryRemoveRepo(repoRoot, out errorMessage))

```

Comment:
This is the line which is 'cleaning up' missing repositories _on read_. I don't think we should be clearing away entries on read? A 'clean up registrations' message would be more explicit, and could also do the check (`volume exists && repo not exists`) listed in open question 2.

Reply:
It would still be nice to clean up old entries without requiring user interaction, and so your suggestion here:

> and could also do the check (volume exists && repo not exists) listed in open question 2.

Sounds like a good option.